lightning-invoice/src/payment.rs

```diff
@@ -131,17 +131,14 @@ use std::sync::Mutex;
 use std::time::{Duration, SystemTime};
 
 /// A utility for paying [`Invoice]`s.
-pub struct InvoicePayer<P: Deref, R, S: Deref, L: Deref, E>
+pub struct InvoicePayer<P: Deref, R: Router<S>, S: routing::Score, Sc: Deref<Target=LockableScore<S>>, L: Deref, E: EventHandler>
```

Comment:
Probably could dispense with `LockableScore` entirely and use `Mutex` directly. Would just require an `unwrap` after `lock` in the code.

Reply:
The bindings don't have any logic for exposing a native `Mutex`, so we have to wrap it :/